### DISCUSSION TITLE
problems with curl probe on Linux CentOS7 with 2.8.2

### DISCUSSION THREAD

**Discussion by dougstarfish** [oetiker/SmokePing#338] [Q&A]
I'm not sure what's going wrong, since I can run curl by hand and it looks okay. Here's what the command looks like after all the extra args are integrated. There's nothing I can see in the perl module that would make me think this would not match as shown below:

/bin/curl "-m" "2" "-w" "Time: %{time_total} DNS time: %{time_namelookup} Redirect time: %{time_redirect}\\n" "-2"  "-4" "--head" "-o" "/tmp/x" --silent http://lists.will.to:8888/pi/
Time: 0.317 DNS time: 0.257 Redirect time: 0.000

(i just took the strace output and removed the ,'s, so ignore all the superfluous quoting). I tried with and without the output redirection, and with --get as well as --head, but for some reason the rrd graphs are still coming up with 100% packet loss.

![n1qoInjQ0SKMBH3v](https://user-images.githubusercontent.com/33408440/169043936-e87dd8fa-3ac6-4213-a1e3-6d61ef05cf39.png)

Also, the debug settings don't seem to work as I would expect. (things end up in 2 places, and there's not a lot of debugging, only extra info at Startup)

I tried setting debug in the config, but not seeing much in the log.
+ Curl

binary = /bin/curl
forks = 2
offset = 20%

expect = Status: green
extraargs = -4 --head -o /dev/null --silent
extrare = / /
follow_redirects = yes
include_redirects = yes
insecure_ssl = 1
#interface = eth0
pings = 5
ssl2 = 1
timeout = 2

---

The fping probe is working fine, just curl that I can't seem to get to work since I want to check some latency on an application port.

--- debug doesn't seem to be outputing much ---
smokeping.conf:
# specify this to get syslog logging
syslogfacility = local6
syslogpriority = debug

rsyslog.conf:
local6.*                                        /var/log/smokeping.log

# tail /var/log/smokeping.log
May 16 17:29:55 dh-sfdemo smokeping[24476]: Entering multiprocess mode.
May 16 17:29:55 dh-sfdemo smokeping[24476]: Child process 24477 started for probe Curl.
May 16 17:29:55 dh-sfdemo smokeping[24477]: Curl: probing 2 targets with step 60 s and offset 12 s.
May 16 17:29:55 dh-sfdemo smokeping[24476]: Child process 24478 started for probe FPing.
May 16 17:29:55 dh-sfdemo smokeping[24476]: All probe processes started successfully.
May 16 17:29:55 dh-sfdemo smokeping[24478]: FPing: probing 1 targets with step 60 s and offset 23 s.
May 16 17:32:28 dh-sfdemo smokeping[24476]: Got TERM signal, terminating child processes.
May 16 17:32:28 dh-sfdemo smokeping[24477]: got TERM signal, terminating.
May 16 17:32:28 dh-sfdemo smokeping[24478]: got TERM signal, terminating.
May 16 17:32:28 dh-sfdemo smokeping[24476]: All child processes successfully terminated, exiting.
[root@dh-sfdemo Ping]# date
Tue May 17 21:29:40 EDT 2022

some messages seem to be bypassing local6 directive too.

[root@dh-sfdemo Ping]# grep -i curl /var/log/messages | tail
May 16 18:16:09 dh-sfdemo smokeping: Child process 7629 started for probe Curl.
May 16 18:16:09 dh-sfdemo smokeping: Curl: probing 2 targets with step 60 s and offset 12 s.Child process 7630 started for probe FPing.
May 16 18:17:06 dh-sfdemo smokeping: Child process 7967 started for probe Curl.
May 16 18:17:06 dh-sfdemo smokeping: Curl: probing 2 targets with step 60 s and offset 12 s.
May 16 18:36:08 dh-sfdemo smokeping: Child process 14326 started for probe Curl.
May 16 18:36:08 dh-sfdemo smokeping: Curl: probing 2 targets with step 60 s and offset 12 s.
May 16 18:37:08 dh-sfdemo smokeping: Child process 14674 started for probe Curl.
May 16 18:37:08 dh-sfdemo smokeping: Curl: probing 2 targets with step 60 s and offset 12 s.
May 16 18:39:09 dh-sfdemo smokeping: Child process 15336 started for probe Curl.
May 16 18:39:09 dh-sfdemo smokeping: Curl: probing 2 targets with step 60 s and offset 12 s.

**Comment by moetiker** [comment]
where is the config 
```
urlformat = http://%host%/ # mandatory
````

**Reply by dougstarfish** [reply]
+++ listsHttp

title = lists http
urlformat = http://%host%:8888/ # mandatory
host = lists.will.to

+++ piHttp

title = pi http proxy
urlformat = http://%host%:8888/pi/ # mandatory
host = lists.will.to

---
I also confirmed via strace that this is evaluating, expanding, and executing curl correctly.

"/bin/curl", "-m", "2", "-w", "Time: %{time_total} DNS time: %{time_namelookup} Redirect time: %{time_redirect}\n", "-2", "-4", "--head", "-o", "/tmp/x", "--silent", "http://lists.will.to:8888/pi/"

(I tried with and without -o, with and without --silent, and with and without --head or --get, but no combination made a difference.)

**Comment by moetiker** [comment]
Could be the :8888 port syntax... have you tried a example on port 80 to see if that's the problem or bug?

**Comment by dougstarfish** [comment]
It's a good thought, but doesn't seem like it. I created a new service with no port and same empty graph.
Also, I can see with strace that curl is execution and returning properly formatted result per the template, so I'm not sure what's going on.
